bitsail-connectors/bitsail-connectors-legacy/bitsail-connector-jdbc/src/main/java/com/bytedance/bitsail/connector/legacy/jdbc/source/JDBCInputFormat.java

```diff
@@ -482,17 +484,13 @@ private void validateParameters() throws ExecutionException, InterruptedExceptio
       LOG.info("Init sql is " + initSql);
     }
 
-    List<ColumnInfo> columns = inputSliceConfig.getNecessaryOption(JdbcReaderOptions.COLUMNS, JDBCPluginErrorCode.REQUIRED_VALUE);
```

Comment:
Actually I think it's better to support two kinds of schema setting:
1. If user does not define `JdbcReaderOptions.COLUMNS`, we can use `getColumnInfoAutomatically'.
2. If user has defined `JdbcReaderOptions.COLUMNS`, we can just use the column infos.

You can use `inputSliceConfig.fieldExist(JdbcReaderOptions.COLUMNS)` to check.

Reply:
@BlockLiu Thanks for your suggestion. i'll do it.